playwright/src/main/java/com/microsoft/playwright/impl/FrameImpl.java

```diff
@@ -1151,4 +1182,20 @@ protected void handleEvent(String event, JsonObject params) {
       internalListeners.notify(InternalEventType.NAVIGATED, params);
     }
   }
+
+  protected double timeout(Double timeout) {
+    TimeoutSettings timeoutSettings = page.timeoutSettings;
```

Comment:
Couldn't `page` be null? I think it is in some rare circumstances. On the other hand, any page should have non-null `timeoutSettings`.